reference/7.0/Microsoft.PowerShell.Core/About/about_Pwsh.md

```diff
@@ -261,6 +261,20 @@ To set up `pwsh` as the login shell on UNIX-like operating systems:
   ```sh
   chsh -s /usr/bin/pwsh
   ```
+- If you are using Homebrew on Linux or macOS, you will need to update your [$PROFILE](/powershell/module/microsoft.powershell.core/about/about_profiles) and let Homebrew adjust your environment variables. Depending on your environment, this could be achieved by adding the following code to `~/.config/powershell/profile.ps1`:
+
+  ```PowerShell
+  $(
+    if (Test-Path -PathType Leaf ~/.linuxbrew/bin/brew)
+      {~/.linuxbrew/bin/brew shellenv}
+    elseif (Test-Path -PathType Leaf /home/linuxbrew/.linuxbrew/bin/brew)
+      {/home/linuxbrew/.linuxbrew/bin/brew shellenv}
+    elseif (Test-Path -PathType Leaf /opt/homebrew/bin/brew)
+      {/opt/homebrew/bin/brew shellenv}
+    elseif (Test-Path -PathType Leaf /usr/local/bin/brew)
+      {/usr/local/bin/brew shellenv}
```

Comment:
i'm not sure what this doing - `brew shellenv` emits a set of strings which look like `export HOMEBREW_PREFIX="/usr/local"` which isn't going to work on PowerShell at all. `export` is not part of the PowerShell language, moreover, there may be syntax that is completely foreign to PowerShell. Piping the following to `Invoke-Expression` will just produce an error:
```powershell
PS> brew shellenv | invoke-expression
export: The term 'export' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
export: The term 'export' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
export: The term 'export' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
export: The term 'export' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
export: The term 'export' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
export: The term 'export' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
export: The term 'export' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
```

Reply:
did this work for you?

Reply:
Have you updated Homebrew to the latest version? `brew shellenv` requires you to have updated to revision https://github.com/Homebrew/brew/commit/13314ca3eb58df5e6ea9d9e3417f06e9ee08eb74

Reply:
my brew seems up to date.
```powershell
PS> brew --version
Homebrew 3.3.6
Homebrew/homebrew-core (git revision 4afc3cc0c40; last commit 2021-12-03)
Homebrew/homebrew-cask (git revision d703802a25; last commit 2021-12-03)
```
could you please provide the output of `brew shellenv` for your system, that might help me understand better.

Reply:
ok - I see that the master branch has support in shellenv.sh for pwsh, but stable does not seem to have this code. I'm not sure that this is appropriate until stable has this

Reply:
also, the code in brew seems overly complicated, 
`echo "[System.Environment]::SetEnvironmentVariable('HOMEBREW_PREFIX','${HOMEBREW_PREFIX}',[System.EnvironmentVariableTarget]::Process)"`
could have been much easier written as
`echo "\$Env:HOMEBREW_PREFIX=\"${HOMEBREW_PREFIX\""`
behaviorally, there should be no difference

Reply:
> could you please provide the output of `brew shellenv` for your system, that might help me understand better.

```
PS /Users/jp> /opt/homebrew/bin/brew shellenv
[System.Environment]::SetEnvironmentVariable('HOMEBREW_PREFIX','/opt/homebrew',[System.EnvironmentVariableTarget]::Process)
[System.Environment]::SetEnvironmentVariable('HOMEBREW_CELLAR','/opt/homebrew/Cellar',[System.EnvironmentVariableTarget]::Process)
[System.Environment]::SetEnvironmentVariable('HOMEBREW_REPOSITORY','/opt/homebrew',[System.EnvironmentVariableTarget]::Process)
[System.Environment]::SetEnvironmentVariable('HOMEBREW_SHELLENV_PREFIX','/opt/homebrew',[System.EnvironmentVariableTarget]::Process)
[System.Environment]::SetEnvironmentVariable('PATH',$('/opt/homebrew/bin:/opt/homebrew/sbin:'+$ENV:PATH),[System.EnvironmentVariableTarget]::Process)
[System.Environment]::SetEnvironmentVariable('MANPATH',$('/opt/homebrew/share/man'+$(if(${ENV:MANPATH}){':'+${ENV:MANPATH}})+':'),[System.EnvironmentVariableTarget]::Process)
[System.Environment]::SetEnvironmentVariable('INFOPATH',$('/opt/homebrew/share/info'+$(if(${ENV:INFOPATH}){':'+${ENV:INFOPATH}})),[System.EnvironmentVariableTarget]::Process)
PS /Users/jp>
```

> ok - I see that the master branch has support in shellenv.sh for pwsh, but stable does not seem to have this code. I'm not sure that this is appropriate until stable has this

I wasn't aware I'm not on the vendors intended version, maybe it's an M1 thing and/or because the arm64 version was in beta for some time. I don't think there is any rush, next Homebrew release will be on Tuesday I suppose (7 days release cycle from what I can see). This PR could be parked for sure.

> also, the code in brew seems overly complicated

There are different coding styles and philosophies I guess. I know how to shorten the code, nevertheless I was following the VSCode recommendation to extend PS code. You may submit a PR to the HB repo but I don't see the point to be honest (I personally don't like to fight for who can write the shortest code in general, but that's me…).
Maybe there is a performance difference with one of the options. If there was, I'd always go for that as loading a users' profile as quickly as possible would be great. I think the more you stay to the .NET internals, the better it becomes, right?